files/build_templates/docker_image_ctl.j2

```diff
@@ -27,6 +27,21 @@ link_namespace() {
 }
 {%- endif %}
 
+function updateSyslogConf()
+{
+    # On multiNPU platforms, change the syslog target ip to docker0 ip to allow logs from containers
+    # running on the namespace to reach the rsyslog service running on the host
+    # Also update the container name  
+    if [[ ($NUM_ASIC -gt 1) ]]; then
+        TARGET_IP=$(ip -o -4 addr list docker0 | awk '{print $4}' | cut -d/ -f1)
+        CONTAINER_NAME="{{docker_container_name}}$DEV"
+        TMP_FILE="/tmp/rsyslog.$CONTAINER_NAME.conf"
```

Comment:
>TMP_FILE="/tmp/rsyslog.$CONTAINER_NAME.conf" [](start = 8, length = 44)

Could you use Dockerfile `ENV` and `docker create --env` to pass the container_name. Then no need to create tmpfile.

Reply:
The tmp file is created after rendering this template /usr/share/sonic/templates/rsyslog-container.conf.j2, which has updated syslog.conf file. I don't think the tmp file can be avoided even if we pass the ENV to the container.

In Multi-NPU platforms it is possible that multi containers can be started at the same time, to avoid container using wrong syslog.conf files I am using the container name in the tmp filename

Reply:
Let's create two ENV:
1. RSYSLOG_SERVER_IP
2. CONTAINER_NAME

Pass the ENV values by command `docker create ... --env RSYSLOG_SERVER_IP=AA.BB.CC.DD ...`

The benefit:
1. not tmp file management
2. works with multiple containers from single image


---
In reply to: [439583041](https://github.com/Azure/sonic-buildimage/pull/4738#discussion_r439583041) [](ancestors = 439583041)

Reply:
@qiluo-msft created a issue to track  [4778](https://github.com/Azure/sonic-buildimage/issues/4778) to improvements suggested in this comment.